components/speedreader/speedreader_service.cc

```diff
@@ -6,33 +6,104 @@
 #include "brave/components/speedreader/speedreader_service.h"
 
 #include "base/feature_list.h"
+#include "base/metrics/histogram_macros.h"
 #include "brave/components/speedreader/features.h"
 #include "brave/components/speedreader/speedreader_pref_names.h"
+#include "brave/components/weekly_storage/weekly_storage.h"
 #include "components/prefs/pref_registry_simple.h"
 #include "components/prefs/pref_service.h"
 
 namespace speedreader {
 
+namespace {
+
+// Note: append-only array! Never remove any existing values, as this array
+// is used to bucket a UMA histogram, and removing values breaks that.
+constexpr std::array<uint64_t, 5> kSpeedReaderToggleBuckets{
+    0,   // 0
+    5,   // >0-5
+    10,  // 6-10
+    20,  // 11-20
+    30,  // 21-30
+         // 30+ => bucket 5
+};
+
+// Note: append-only enumeration! Never remove any existing values, as this enum
+// is used to bucket a UMA histogram, and removing values breaks that.
+enum class EnabledStatus {
+  kUnused,
+  kEverEnabled,
+  kRecentlyUsed,
+  kMaxValue = kRecentlyUsed
+};
+
+constexpr char kSpeedreaderToggleUMAHistogramName[] =
+    "Brave.SpeedReader.ToggleCount";
+
+constexpr char kSpeedreaderEnabledUMAHistogramName[] =
+    "Brave.SpeedReader.Enabled";
+
+void StoreTogglesHistogram(uint64_t toggles) {
+  int bucket = 0;
+  for (const auto& bucket_upper_bound : kSpeedReaderToggleBuckets) {
+    if (toggles > bucket_upper_bound)
+      bucket = bucket + 1;
+  }
+
+  UMA_HISTOGRAM_EXACT_LINEAR(kSpeedreaderToggleUMAHistogramName, bucket, 5);
+}
+
+void StoreSpeedReaderEnabledHistogram(PrefService* prefs,
```

Comment:
It actually stores 2 histograms, one of them is not related to "enabled". So maybe this is `StoreSpeedReaderMetrics`? (or just `RecordHistograms` since we are anyway in speedreader_service.cc)